### DISCUSSION TITLE
Checks are not running. Stuck in "Expected — Waiting for status to be reported"

### DISCUSSION THREAD

**Discussion by mvilanova** [community/community#26148] [Actions]
<p>Checks on push are not running for any PRs and are stuck in “Expected — Waiting for status to be reported”.</p>
<p>Our workflow can be found here: <a href="https://github.com/Netflix/dispatch/blob/develop/.github/workflows/list_and_test.yml" rel="nofollow noopener">https://github.com/Netflix/dispatch/blob/develop/.github/workflows/list_and_test.yml</a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/48e4610b71bd2f0d6fc07ffddbd346c595dc2f57.png" data-download-href="/uploads/short-url/aoPG1rSyjrDk5aTixpjOZ2vmHL9.png?dl=1" title="Screen Shot 2020-03-09 at 9.04.27 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/39573146/181095818-0183f1df-1140-4964-888a-5a00089d3af0.png" alt="Screen Shot 2020-03-09 at 9.04.27 PM.png" data-base62-sha1="aoPG1rSyjrDk5aTixpjOZ2vmHL9" width="690" height="434" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/48e4610b71bd2f0d6fc07ffddbd346c595dc2f57_2_690x434.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/48e4610b71bd2f0d6fc07ffddbd346c595dc2f57_2_1035x651.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/4/48e4610b71bd2f0d6fc07ffddbd346c595dc2f57_2_1380x868.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/48e4610b71bd2f0d6fc07ffddbd346c595dc2f57_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-03-09 at 9.04.27 PM.png</span><span class="informations">1686×1062 112 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by jameswnl** [comment]
<p>seeing same issue with 1 PR</p>
<p>Other PRs are ok.</p>
<p>this PR is adding <code>.flake8</code> file only, is this “ignored”? but then why it is <code>expected</code>?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/mvilanova">@mvilanova</a> ,</p>
<p>Thank you for being here! </p>
<p>Checks on push are not running for any PRs</p>
<p>&gt;&gt; If you create PR from branch B to branch A, ‘pull_request’ event will be triggered not ‘push’ event.</p>
<p>If you have ‘push’ event (not pull_request event) in branch B, while ‘pull_request’ event in branch A, PR will be triggered with branch A workflow file. If you have ‘pull_request’ event in Branch B, it will use branch B workflow file.</p>
<p>Until you merge code to branch A,  push event on branch A will be triggered.</p>
<p>stuck in “Expected — Waiting for status to be reported”</p>
<p>&gt;&gt;Please check your branch protection policy for branch ‘develop’, uncheck the ‘checkruns’ if it exists, close&amp;recreate the PR for a double check. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/dda4d4b6f3a571aca9d45e5ef7836125d3a212a1.png" data-download-href="/uploads/short-url/vCKFvRwbxzbT0lL9mzZizfRkFe9.png?dl=1" title="branchprotectionrule.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181095822-ef5905e2-41b5-4f1a-add8-1dafdd4f251f.png" alt="branchprotectionrule.png" data-base62-sha1="vCKFvRwbxzbT0lL9mzZizfRkFe9" width="690" height="331" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/dda4d4b6f3a571aca9d45e5ef7836125d3a212a1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">branchprotectionrule.png</span><span class="informations">699×336 18.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>If it’s still stuck for checkruns, please let me know your PR link for further investigation.</p>
<p>Hope it helps!</p>

**Reply by samthui** [reply]
For me, closing then reopening the PR solves the issue

**Reply by ssoulless** [reply]
Thanks, that solved mine too

**Comment by mvilanova** [comment]
<p>Looks like we were using the wrong event type. Everything’s working after we changed it to <code>pull\_request</code>. Thank you!</p>

**Comment by mvilanova** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> follow up question, what if branch B lives in a fork? What type of event is triggered E.g. <a href="https://github.com/Netflix/dispatch/pull/92" rel="nofollow noopener">https://github.com/Netflix/dispatch/pull/92</a></p>

**Comment by weide-zhou** [comment]
<p>The event is same with NON-fork.  Only need to mention is, for pull_request from forked repo to base repo, the workflow will be executed on base repo. Please check the official <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories" rel="nofollow noopener">doc</a> for more details. Thanks.</p>

**Comment by anmolchopra69** [comment]
How can we add Details link for these actions.
